### PR TITLE
ANTLR based contact searching

### PR DESCRIPTION
**Pull request by rowanseymour**
So we can use the same "ContactQL" grammar in goflow for dynamic group evaluation

### CODE REVIEW

File: temba/contacts/tests.py [hunk 1]
```diff
@@ -1082,13 +1082,13 @@ def test_contact_search_parsing(self):
 
         # boolean operator precedence is AND before OR, even when AND is implicit
         self.assertEqual(parse_query('will and felix or matt amber', optimize=False), ContactQuery(
-            BoolCombination(BoolCombination.AND,
```
Comment (rowanseymour) [inline temba/contacts/tests.py]: Pretty sure this test is wrong

File: pip-freeze.txt [hunk 2]
```diff
@@ -95,7 +94,7 @@ python-magic==0.4.13
 python-telegram-bot==6.1.0
 pytz==2017.2
 pyyaml==3.12              # via uservoice
-rapidpro-expressions==1.1.10
+rapidpro-expressions==1.2
```
Comment (rowanseymour) [inline pip-freeze.txt]: Had to update rapidpro-expressions so all ANTLR grammars are using the same runtime version (4.7)

File: temba/contacts/tests.py (hunk 1 above)
Comment (nicpottier) [reply temba/contacts/tests.py]: I think generally AND is considered higher precedence as OR, so isnt this correct but the new behavior wrong?

Comment (nicpottier) [reply temba/contacts/tests.py]: NM, you are correct!

**Pull request merged**